packages/swapper/src/swappers/zrx/approveInfinite/approveInfinite.ts

```diff
@@ -15,7 +15,7 @@ export async function approveInfinite(
 ) {
   const adapter: ChainAdapter<ChainTypes.Ethereum> = adapterManager.byChain(ChainTypes.Ethereum)
   const bip32Params = adapter.buildBIP32Params({
-    accountNumber: Number(quote.sellAssetAccountId || quote.sellAssetAccountId) || 0
+    accountNumber: quote.sellAssetAccountId ? Number(quote.sellAssetAccountId) : 0
```

Comment:
`bnOrZero` please

Reply:
if you need a `Number` instance, you can call `toNumber()` on the BN value.